### PR TITLE
Enables Go Modules support (go v1.13, operator-sdk v0.9.0)

### PR DESCRIPTION
**Pull request by clcollins**
This commit enables Go modules support, which requires Go 1.13 and operator-sdk v0.9.0.  Consequently, locks Go language feature support to 1.13 in go.mod, upgrades the operator-sdk from 0.8.1 to 0.9.0 to enable full Go modules support, and fixes the Dockerfile to the Go 1.13 image.

Tested make targets with no apparent issues.

**This change requires moving the aws-account-operator project folder from your $GOPATH, `export GO111MODULE=on`, and downloading the operator-sdk v0.9.0 binary** (if you do not have it separate from Go bin path).  The team will need to make these changes to their local environment when this code is merged.

Files changes (other than deleted /vendor directory):

* deleted:    Gopkg.lock
* deleted:    Gopkg.toml
* modified:   build/Dockerfile
* new file:   go.mod
* new file:   go.sum

Signed-off-by: Christopher Collins <collins.christopher@gmail.com>

### CODE REVIEW

**Comment by clcollins** [comment]
Having some issues with `test-awsfederatedrole` & `test-aws-ou-logic`

Not sure if this is related to the change, or some AWS errors, but for completeness, mentioning here.

File: go.mod
Comment (2uasimojo) [inline go.mod:16]: This plus...

File: go.mod
Comment (2uasimojo) [inline go.mod:54]: ...this equals weird. Why wouldn't we just use `v0.9.0` on L16?

**Review by 2uasimojo — DISMISSED** [review]
Mechanically this looks to be going in the right direction. (Can I just mention how obvious the blatant wrongness of vendored deps becomes when you can submit a PR that nets -4MLOC?)

A couple of minor things in addition to making tests go green:

1. Please update the [operator-sdk reference in the README](https://github.com/openshift/aws-account-operator/blame/master/README.md#L171) to the appropriate version.
2. Does 0.9.0 still have the same syntax with respect to things like `operator-sdk up local`? I know at some point they changed it to `operator-sdk run --local`. If there are discrepancies there, the `make` targets may need to be updated appropriately.

File: go.mod
Comment (clcollins) [reply go.mod:54]: Yeah, this was weird to me as well.  It was in the documentation to add the items below `// Pinned to kubernetes-1.13.1` coming up from v0.8.2, which prevented conflicts with the k8s client-runtime, which from what I understand, stems from an ideological argument between K8S maintainers and the Go maintainers over semantic versioning.

File: go.mod
Comment (clcollins) [reply go.mod:16]: See below

**Comment by clcollins** [comment]
@2uasimojo: 

> * Please update the [operator-sdk reference in the README](https://github.com/openshift/aws-account-operator/blame/master/README.md#L171) to the appropriate version.

Done!

> * Does 0.9.0 still have the same syntax with respect to things like `operator-sdk up local`? I know at some point they changed it to `operator-sdk run --local`. If there are discrepancies there, the `make` targets may need to be updated appropriately.

It appears to still have the same API as earlier versions, at least for the pieces we are using.

**Comment by 2uasimojo** [comment]
Thanks for the updates @clcollins.

I'll note that [this commit](https://github.com/openshift/aws-account-operator/pull/466/commits/3a02c2b58465461379cb36464fc31c3dfffd0b42) doesn't really appear to be doing what its title says (the osdk lines weren't touched).

Will continue watching for CI to go green.

**Comment by clcollins** [comment]
@2uasimojo:

> I'll note that [this commit](https://github.com/openshift/aws-account-operator/pull/466/commits/3a02c2b58465461379cb36464fc31c3dfffd0b42) doesn't really appear to be doing what its title says (the osdk lines weren't touched).

The lines that were changed were the ones documented by the operator-sdk as needing to be added when upgrading to v0.9.0, specifically as a dependency for v0.9.0.

Ie: I'm fixing the "replaces" lines, as documented for v0.9.0.  :)

**Comment by 2uasimojo** [comment]
> Ie: I'm fixing the "replaces" lines, as documented for v0.9.0. :)

Ah, thanks for the explanation.

Since this is CI green, and looks mechanically sound, I'm happy with it. I didn't pull it down and test it (not sure I'm going to get a chance) so I'll avoid stamping it officially. But lgtm.

**Comment by clcollins** [comment]
Ok, all make tests are working fine now - issues with my local environment not with the code after the upgrades.

**Comment by rogbas** [comment]
@rogbas @iamkirkbater

**Comment by rogbas** [comment]
LGTM, merge it on Oct 5th

/hold

**Pull request merged**
